node/syncblock.go

```diff
@@ -334,10 +360,12 @@ func (localNode *LocalNode) getNeighborsMajorityBlockHashByHeight(height uint32,
 		}
 
 		counter := make(map[common.Uint256]int)
+		headers := make(map[common.Uint256]*block.Header)
 		totalCount := 0
 		allHeaders.Range(func(key, value interface{}) bool {
 			if header, ok := value.(*block.Header); ok && header != nil {
 				counter[header.Hash()]++
+				headers[header.Hash()] = value.(*block.Header)
```

Comment:
Need to check header signature before set this because hash does not depend on signature.